manifests/init.pp

```diff
@@ -22,60 +22,68 @@
   include ::yum::updatesd
 
   validate_absolute_path($config_path)
-  validate_string($config_owner)
-  validate_string($config_group)
-  validate_re($config_mode, '^[0-7]{4}$',
-    "yum::config_mode is <${config_mode}> and must be a valid four digit mode in octal notation.")
 
-  validate_string($repos_d_owner)
-  validate_string($repos_d_group)
-  validate_re($repos_d_mode, '^[0-7]{4}$',
-    "yum::repos_d_mode is <${repos_d_mode}> and must be a valid four digit mode in octal notation.")
+  case $distroverpkg {
+    true, 'true':   { $distroverpkg_bool = true }  # lint:ignore:quoted_booleans
+    false, 'false': { $distroverpkg_bool = false } # lint:ignore:quoted_booleans
+    default:        { fail("yum::distroverpkg is not a boolean. It is <${distroverpkg}>.") }
+  }
+
+  validate_string(
+    $config_owner,
+    $config_group,
+    $repos_d_owner,
+    $repos_d_group,
+  )
 
   if $proxy != undef {
     validate_string($proxy)
   }
 
-  if $installonly_limit != undef {
-    if is_string($installonly_limit) == false and is_integer($installonly_limit) == false {
-      $installonly_limit_type = type3x($installonly_limit)
-      fail("yum::installonly_limit is <${installonly_limit}> with type ${installonly_limit_type} and must be a string or an integer")
-    }
+  if $pkgpolicy != undef {
+    validate_re($pkgpolicy, '^(newest|last)$',
+      "yum::pkgpolicy must be <newest> or <last>. It is <${pkgpolicy}>.")
   }
 
-  if type3x($manage_repos) == 'string' {
-    $manage_repos_bool = str2bool($manage_repos)
-  } else {
-    $manage_repos_bool = $manage_repos
+  validate_re($config_mode, '^[0-7]{4}$',
+    "yum::config_mode is not a file mode in octal notation. It is <${config_mode}>.")
+
+  validate_re($repos_d_mode, '^[0-7]{4}$',
+    "yum::repos_d_mode is not a file mode in octal notation. It is <${config_mode}>.")
+
+
+  if $installonly_limit != undef {
+    case type3x($installonly_limit) {
+      'integer': { $installonly_limit_int = $installonly_limit }
+      'string':  { $installonly_limit_int = floor($installonly_limit) }
+      default:   { fail("yum::installonly_limit is not an integer nor stringified integer. It is <${installonly_limit}>.")}
+    }
   }
-  validate_bool($manage_repos_bool)
 
-  case $manage_repos_bool {
-    true: {
+  case $manage_repos {
+    true, 'true': { # lint:ignore:quoted_booleans
```

Comment:
Using ```str2bool()``` here doesn't catch wrong data types and would need more additional code to catch them too. Therefore I see no enhancement in using it.

But if you think it would be easier to read, I can change that of course.